### PR TITLE
contracts: Add VRF load testing contracts

### PR DESCRIPTION
**Pull request by Nic0S**
Add VRF consumer contracts for v1 and v2 that create multiple
randomness requests in a single transaction for the purposes
of load testing

### CODE REVIEW

File: contracts/src/v0.8/tests/VRFLoadTestExternalSubOwner.sol [hunk 1]
```diff
@@ -0,0 +1,56 @@
...
+  }
+
+  function transferOwnership(address newOwner) external onlyOwner {
+    s_owner = newOwner;
+  }
+
+  modifier onlyOwner() {
+    require(msg.sender == s_owner);
+    _;
+  }
```
Comment (makramkd) [inline contracts/src/v0.8/tests/VRFLoadTestExternalSubOwner.sol:55]: I think you can replace these and the `address s_owner` with an `import "Ownable.sol"` and inheriting it. Looks like it's not available for 0.8 though so you'd have to copy/paste it into the folder.

File: contracts/src/v0.8/tests/VRFLoadTestOwnerlessConsumer.sol [hunk 2]
```diff
@@ -0,0 +1,50 @@
...
+
+import "../VRFConsumerBase.sol";
+import "../interfaces/ERC677ReceiverInterface.sol";
+
+/**
+ * @title The VRFLoadTestOwnerlessConsumer contract.
+ * @notice Allows making many VRF V1 randomness requests in a single transaction for load testing.
+ */
+contract VRFLoadTestOwnerlessConsumer is VRFConsumerBase, ERC677ReceiverInterface {
+  // The price of each VRF request.
```
Comment (makramkd) [inline contracts/src/v0.8/tests/VRFLoadTestOwnerlessConsumer.sol:12]: Would be helpful for newcomers to also state that this price is in Juels, the lowest denomination of LINK: `// The price of each VRF request in Juels, the lowest denomination of LINK. 1 LINK = 1e18 Juels.` or something similar.

File: core/scripts/vrfv1/main.go [hunk 3]
```diff
@@ -84,6 +96,25 @@ func main() {
 		helpers.PanicErr(err)
 		fmt.Printf("Ownerless Consumer: %s TX: %s\n",
 			consumerAddr, helpers.ExplorerLink(chainID, tx.Hash()))
+	case "loadtest-ownerless-consumer-deploy":
+		cmd := flag.NewFlagSet("loadtest-ownerless-consumer-deploy", flag.ExitOnError)
+		coordAddr := cmd.String("coordinator-address", "", "address of VRF coordinator")
+		linkAddr := cmd.String("link-address", "", "address of link token")
+		priceStr := cmd.String("price", "", "the price of each VRF request")
```
Comment (makramkd) [inline core/scripts/vrfv1/main.go:103]: `"the price of each VRF request in Juels"` would be a more specific comment.

File: core/scripts/vrfv1/main.go [hunk 4]
```diff
@@ -84,6 +96,25 @@ func main() {
...
+		helpers.ParseArgs(cmd, os.Args[2:], "coordinator-address", "link-address")
+		price, ok := big.NewInt(0).SetString(*priceStr, 10)
+		if !ok {
+			panic(fmt.Sprintf("failed to parse price amount: %s", *priceStr))
+		}
```
Comment (makramkd) [inline core/scripts/vrfv1/main.go:108]: I think you can use `decimal.RequireFromString(priceStr).BigInt()` here to save on boilerplate.

File: contracts/src/v0.8/tests/VRFLoadTestExternalSubOwner.sol [hunk 5]
```diff
@@ -0,0 +1,56 @@
...
+
+  function fulfillRandomWords(uint256, uint256[] memory) internal override {
+    s_responseCount++;
+  }
+
+  function requestRandomWords(
+    uint64 _subId,
+    uint32 _callbackGasLimit,
+    uint16 _requestConfirmations,
+    uint32 _numWords,
```
Comment (makramkd) [inline contracts/src/v0.8/tests/VRFLoadTestExternalSubOwner.sol:33]: Are these arguments necessary to have in this method? Are we thinking we'd vary `_callbackGasLimit`, `_requestConfirmations`, etc. for a load test, or is it less boilerplate-y to have them as constants like `COORDINATOR` and `LINK` if they're not gonna change much. We'd probably want to load test on diff keyhashes though, so makes sense to have the keyhash as an argument.

**Comment by makramkd** [comment]
Do we want to also update the `README.md` for each respective `main.go` helper script you've updated?

**Comment by Nic0S** [comment]
Nah I don't think this is README worthy, it's a pretty niche feature. The current READMEs don't document every command, just the basic deploy-request randomness flow

**Comment by cl-homu** [comment]
:umbrella: The latest upstream changes (presumably b00c14524e585f84f59b1b5f0af2b28698860cf7) made this pull request unmergeable. Please [resolve the merge conflicts](https://rustc-dev-guide.rust-lang.org/git.html#rebasing-and-conflicts).

**Review by makramkd — DISMISSED** [review]
🚢

**Review by makramkd — APPROVED** [review]

**Pull request merged**